src/main/java/javaslang/collection/HashMap.java

```diff
@@ -389,6 +405,16 @@ public boolean isTraversableAgain() {
     }
 
     @Override
+    public HashMap<K, V> put(java.util.Map<? extends K, ? extends V> map) {
```

Comment:
It is not a good idea to have this put method. In fact it is a _merge_ of maps which needs to handle collisions. We already have it.

Also I do not want to add specific java.util.* methods to the Javaslang library. For conversion we have the `ofAll` factory methods and the `toJavaXxx()` methods.

Please move the body of the `put` methods to the `ofAll(java.util.Map)` methods. We then can do the following:

```java
java.util.Map<...> map1 = ...;
javaslang.collection.Map<...> map2 = ...;
map2.merge(HashMap.ofAll(map1));
```